src/Features/CSharp/Portable/Completion/KeywordRecommenders/ThrowKeywordRecommender.cs

```diff
@@ -40,7 +38,7 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
             if (context.TargetToken.Kind() is SyntaxKind.QuestionToken or
                 SyntaxKind.ColonToken)
             {
-                return context.TargetToken.Parent.Kind() == SyntaxKind.ConditionalExpression;
+                return context.TargetToken.Parent?.Kind() == SyntaxKind.ConditionalExpression;
```

Comment:
```suggestion
                return context.TargetToken.Parent.IsKind(SyntaxKind.ConditionalExpression)
```

or (it needs `using Microsoft.CodeAnalysis.CSharp.Extensions;`)

```suggestion
                return context.TargetToken.IsParentKind(SyntaxKind.ConditionalExpression)
```

Reply:
In the first suggestion you drop null check for `Parent` making code less safe. It is possible to have `context.TargetToken.Parent?.IsKind(SyntaxKind.ConditionalExpression) ?? false` (`?? false` is required because of `?.` operator), but this is less readable than the thing we currently have.

For the second suggestion, well, `SyntaxToken` doesn't have extension method `IsParentKind`, it is only defined for `SyntaxNode`

Reply:
@DoctorKrolic

> In the first suggestion you drop null check for Parent making code less safe.

`IsKind` is an extension method and it checks for null:

https://github.com/dotnet/roslyn/blob/32e8d1804d6a31ac00cbb82a50be777eb54ad0b3/src/Compilers/CSharp/Portable/CSharpExtensions.cs#L47-L49

So the code is still safe.

Reply:
I don't usually expect invocations on `null` to be safe, but ok...

Reply:
It's totally safe. `IsKind` is just a static method call where the first parameter can be null, and the static method handles null correctly.

See [simplified example](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4AEBMBGAWAChiAFAU2QGcB7GAAgAd6BeemVAG04G5is8ATgAUjAHQBJKgHFyMACaVhshZTEBZCJ3IBKHXxJEsAZnq56FanWIBvYvQdnTl2jGVzFyegHMPlHawAfPQqnqw+fsgG9o4mIZHxql42EQg89AC+xFmGcXKoYImetjEOmtpwpfQAYuSQFdnE/KYCAGxmOPQAogAeCHJUUHRUJUSOTmZ47cA0NJz0UqFKCAAWUFQWlK5MW3RwRZQRSTpVdmPjsQDsO1YwAPxiS14sbL5JBuM5GUA)